### DISCUSSION TITLE
Internal error, please try again later. If the problem persists, please contact our support.

### DISCUSSION THREAD

**Discussion by meschnigm** [tillsteinbach/WeConnect-mqtt#126] [Q&A] [answered]
Hello Till,

I was successfully installed WeConnecct-mqtt on my raspberry (debian 10 buster).  When I start the script with `weconnect-mqtt --mqttbroker 192.178.168.86&` it is running in background and I receive one time the data of my car.

`ps | grep mqtt` shows a running process with ID 
`17611 pts/1    00:00:01 weconnect-mqtt`

In FHEM I subscribed all topics with 
`weconnect/vehicles/WVWZZZAUZFW809xxx`

**Now I struggling with the following:**
1) weconnect-mqtt reports an internal failure
    Internal error, please try again later. If the problem persists, please contact our support.

2) I only receive the data once - first I thought this is normal and was starting to configure a cronjob which runs the script every 5 minutes. But as I saw this is not the idea of the script. Is the internal failure the reason I don't receive continuous updates from my vehicle?

3) I found the description `[to setup the script in a service and start the script during system startup ](https://github.com/tillsteinbach/WeConnect-mqtt/tree/main/autostart/systemd)` 

`sudo systemctl status weconnect-mqtt
● weconnect-mqtt.service - WeConnect-mqtt
   Loaded: loaded (/etc/systemd/system/weconnect-mqtt.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2022-06-08 13:40:46 CEST; 4h 28min ago
  Process: 7274 ExecStart=/usr/bin/python -m weconnect_mqtt --username user --password geheim--mqttbroker 192.178.168.86 --mqtt-username admin --mqtt-password geheim
 Main PID: 7274 (code=exited, status=1/FAILURE)

Jun 08 13:40:45 raspberrypi systemd[1]: Started WeConnect-mqtt.
Jun 08 13:40:46 raspberrypi python[7274]: /usr/bin/python: **No module named weconnect_mqtt**
Jun 08 13:40:46 raspberrypi systemd[1]: weconnect-mqtt.service: Main process exited, code=exited, status=1/FAILURE
Jun 08 13:40:46 raspberrypi systemd[1]: weconnect-mqtt.service: Failed with result 'exit-code'.
`

`cat /etc/systemd/system/weconnect-mqtt.service`
```
[Unit]
Description=WeConnect-mqtt
After=network-online.target

[Service]
Type=simple
# change the username and commandline parameters here:
User=pi
ExecStart=/usr/bin/python -m weconnect_mqtt --username user --password geheim --mqttbroker 192.178.168.86 --mqtt-username admin --mqtt-password geheim--prefix weconnect
# Restart=on-failure

[Install]
WantedBy=multi-user.target

```

`ls -l /etc/systemd/system/weconnect-mqtt.service`
`-rw-r--r-- 1 pi pi 402 Jun  6 22:21 /etc/systemd/system/weconnect-mqtt.service`

Thank you in advance for our support.
BR Manuel

**Comment by tillsteinbach** [comment]
Where do you see the internal failure?
I would recommend to increase the output with -vv
Regarding the Startup script: it looks like the script is using a different python Version where the module is not installed. Maybe you have to exchange python with python3 in ExecStart?

**Reply by meschnigm** [reply]
The hint with python3 helped and also the -vv is helping to answer your next question where the internal failure is visible.

`pi@raspberrypi:~ $ weconnect-mqtt -vv --mqttbroker 192.178.168.86&
[1] 2879
pi@raspberrypi:~ $ 2022-06-09T18:32:21+0200:INFO:weconnect_mqtt_base:WeConnect-mqtt 0.34.0 (using WeConnect-python 0.40.0)
2022-06-09T18:32:21+0200:INFO:weconnect_mqtt_base:Connect to WeConnect
2022-06-09T18:32:22+0200:INFO:weconnect_mqtt_base:Connected to MQTT broker
2022-06-09T18:32:22+0200:INFO:weconnect_mqtt_base:Update data from WeConnect
2022-06-09T18:32:25+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute brandCode with value V
2022-06-09T18:32:26+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute error with value {'message': 'Internal error', 'info': 'Internal error, please try again later. If the problem persists, please contact our support.', 'errorTimeStamp': '2022-06-09T16:32:25Z', 'code': 4001, 'group': 2, 'retry': True} in domain readiness`

Unfortenately new data I receive only ones during startup - no cyclic update.
BR Manuel

**Comment by tillsteinbach** [comment]
The error message you can ignore. It is because auf an item that was unknown. The message will be gone with the next release.
How long do you wait for new updates to arrive?

**Comment by meschnigm** [comment]
Beccause of weconnectUpdateInterval_s | 300 I expected an update after 5 Minutes. 

`sudo systemctl status weconnect-mqtt`
shows Active: active (running) during the running time > 5 Minutes.
BR Manuel

**Reply by tillsteinbach** [reply]
Ok, that is correct. And with -vv you do not see that an update is being done after 300seconds?

**Reply by meschnigm** [reply]
pi@raspberrypi:~ $ 2022-06-09T21:01:10+0200:INFO:weconnect_mqtt_base:WeConnect-mqtt 0.34.0 (using WeConnect-python 0.40.0)
2022-06-09T21:01:10+0200:INFO:weconnect_mqtt_base:Connect to WeConnect
2022-06-09T21:01:10+0200:INFO:weconnect_mqtt_base:Connected to MQTT broker
2022-06-09T21:01:10+0200:INFO:weconnect_mqtt_base:Update data from WeConnect
2022-06-09T21:01:13+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute brandCode with value V
2022-06-09T21:01:13+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute error with value {'message': 'Internal error', 'info': 'Internal error, please try again later. If the problem persists, please contact our support.', 'errorTimeStamp': '2022-06-09T19:01:13Z', 'code': 4001, 'group': 2, 'retry': True} in domain readiness
2022-06-09T21:05:24+0200:INFO:weconnect_mqtt_base:Update data from WeConnect
2022-06-09T21:05:25+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute brandCode with value V
2022-06-09T21:05:25+0200:WARNING:vehicle:/vehicles/WVWZZZAUZFW809xxx: Unknown attribute error with value {'message': 'Internal error', 'info': 'Internal error, please try again later. If the problem persists, please contact our support.', 'errorTimeStamp': '2022-06-09T19:05:25Z', 'code': 4001, 'group': 2, 'retry': True} in domain readiness

**Reply by meschnigm** [reply]
Looks like the script is executed but in FHEM - I can't see any update.
Is this just an issue with my FHEM configuration?
```defmod MQTT2_MQTT2_SERVER MQTT2_DEVICE MQTT2_SERVER
attr MQTT2_MQTT2_SERVER alias Golf 1,4 GTEBMHY110TSI D6
attr MQTT2_MQTT2_SERVER devicetopic weconnect/vehicles/WVWZZZAUZFW809xxx```

**Reply by tillsteinbach** [reply]
So you see, it is doing the updates. Ahh, maybe the misunderstanding is: by default the topics are only published when something is changing and not every 300s. So maybe there is no change?
What is interesting is:
`brandCode with value V`
I have never seen the brandCode attribute before. What kind of vehicle is this?

**Reply by meschnigm** [reply]
It is an 2015 Golf GTE.

Ahh, maybe the misunderstanding is: by default the topics are only published when something is changing and not every 300s. So maybe there is no change?
--> ok - this is an explanation for what I see. 
Thank you for this hint.

with verbose 5 of the MQTT device is see the follwoing in FHEM log.
```pi@raspberrypi:~ $ tail -f /opt/fhem/log/fhem-2022-06.log | grep weconnect
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/automation/climatisationTimer/error/timestamp => timestamp
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/automation/chargingProfiles/error/timestamp => timestamp
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/charging/chargeMode/error/timestamp => timestamp
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/lvBattery/lvBatteryStatus/error/timestamp => timestamp
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/vehicleHealthWarnings/warningLights/error/timestamp => timestam                 p
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/oilLevel/oilLevelStatus/error/timestamp => timestamp
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/vehicles/WVWZZZAUZFW809xxx/domains/batterySupport/batterySupportStatus/error/timestamp => timestam                 p
2022.06.09 21:34:07 4: MQTT2_DEVICE_Parse: MQTT2_MQTT2_SERVER weconnect/0/mqtt/weconnectUpdated => weconnectUpdated
```

**Reply by tillsteinbach** [reply]
Looks like data is coming. You can also change the behavior with adding `--republish-on-update`. Then you get all the topics published with every update. But of course this generates a lot of unnecessary traffic in FHEM.

**Comment by meschnigm** [comment, marked as answer]
The autostart was not working - I got:
```pi@raspberrypi:~ $ sudo systemctl status weconnect-mqtt
● weconnect-mqtt.service - WeConnect-mqtt
   Loaded: loaded (/etc/systemd/system/weconnect-mqtt.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2022-06-09 22:42:49 CEST; 1min 9s ago
  Process: 485 ExecStart=/usr/bin/python3 -m weconnect_mqtt --username user  --password geheim --mqttbroker 192.178.168.86 --mqtt-username admin --mqtt-password admin_fh
 Main PID: 485 (code=exited, status=1/FAILURE)

Jun 09 22:42:49 raspberrypi python3[485]:     sock = self._create_socket_connection()
Jun 09 22:42:49 raspberrypi python3[485]:   File "/home/pi/.local/lib/python3.7/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
Jun 09 22:42:49 raspberrypi python3[485]:     return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
Jun 09 22:42:49 raspberrypi python3[485]:   File "/usr/lib/python3.7/socket.py", line 727, in create_connection
Jun 09 22:42:49 raspberrypi python3[485]:     raise err
Jun 09 22:42:49 raspberrypi python3[485]:   File "/usr/lib/python3.7/socket.py", line 716, in create_connection
Jun 09 22:42:49 raspberrypi python3[485]:     sock.connect(sa)
Jun 09 22:42:49 raspberrypi python3[485]: OSError: [Errno 101] Das Netzwerk ist nicht erreichbar
Jun 09 22:42:49 raspberrypi systemd[1]: weconnect-mqtt.service: Main process exited, code=exited, status=1/FAILURE
Jun 09 22:42:49 raspberrypi systemd[1]: weconnect-mqtt.service: Failed with result 'exit-code'.
p```



I added `ExecStartPre=/bin/sleep 30` in weconnect-mqtt.service file to solve this.

Over and Out
Thank you

**Reply by tillsteinbach** [reply]
That is interesting because the service specified that it should run after the network is up. Anyway, good to see that it is working now!
